### PR TITLE
chore: apply spotless

### LINKED ISSUE #1716: Apply spotless plugin
Apply the spotless plugin to improve code quality and refactor code according to java code style guideline.

**Would you like to work on the issue?**

- [x] Yes
- [ ] No
- Other:

### PR DESCRIPTION
**Pull request by adityastic**
Fixes #1716 #1922

### CODE REVIEW

**Comment by adityastic** [comment]
@iamareebjamal @ShridharGoel

**Comment by iamareebjamal** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 6
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/fossasia/open-event-orga-app/pullRequest?prid=4677093&bid=15648597)

File: app/build.gradle [hunk 1]
```diff
@@ -109,6 +112,16 @@ android {
     }
 }
 
+spotless {
+    java {
+        removeUnusedImports()
+        indentWithSpaces(2)
```
Comment (iamareebjamal) [inline app/build.gradle:121]: Why 2 spaces?

**Comment by iamareebjamal** [comment]
Please add spotless check in circle CI and remove from travis

File: app/build.gradle (hunk 1 above)
Comment (adityastic) [reply app/build.gradle:121]: it was not matching the android studio's formatting standards with default 1 space.

eg: 
1 Space:
```
public class AboutUsFragment extends BaseFragment {
  private AboutUsFragmentBinding binding;
```
2 Space:
```
public class AboutUsFragment extends BaseFragment {
    private AboutUsFragmentBinding binding;
```
I tried to match the default android formatting used by android studio or IntelliJ

**Comment by adityastic** [comment]
> Please add spotless check in circle CI and remove from travis

On it

**Comment by iamareebjamal** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 13
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/fossasia/open-event-orga-app/pullRequest?prid=4677093&bid=15648597)

**Comment by adityastic** [comment]
> Codacy Here is an overview of what got changed by this pull request:
> 
> ```diff
> Issues
> ======
> + Solved 13
>            
> ```
> 
> See the complete overview on [Codacy](https://app.codacy.com/app/fossasia/open-event-orga-app/pullRequest?prid=4677093&bid=15648597)

Perfect, Almost there xD

**Comment by iamareebjamal** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 15
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/fossasia/open-event-orga-app/pullRequest?prid=4677093&bid=15648597)

**Comment by iamareebjamal** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 15
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/fossasia/open-event-orga-app/pullRequest?prid=4677093&bid=15648597)

**Comment by iamareebjamal** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 15
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/fossasia/open-event-orga-app/pullRequest?prid=4677093&bid=15648597)

**Comment by iamareebjamal** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 15
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/fossasia/open-event-orga-app/pullRequest?prid=4677093&bid=15648597)

**Comment by adityastic** [comment]
@iamareebjamal perfect, fixed the ci. let me know if we need to improve anything

File: app/build.gradle (hunk 1 above)
Comment (iamareebjamal) [reply app/build.gradle:121]: OK

File: .circleci/config.yml [hunk 2]
```diff
@@ -19,10 +19,13 @@ jobs:
           key: jars-{{ checksum "build.gradle" }}-{{ checksum  "app/build.gradle" }}
       - run:
           name: Run Tests
-          command: ./gradlew lintPlayStoreDebug androidCheck androidPmd --no-daemon
+          command: ./gradlew lintPlayStoreDebug androidCheck androidPmd --no-daemon --max-workers 2
+      - run:
+          name: Spotless Test
+          command: ./gradlew spotlessJavaCheck
```
Comment (iamareebjamal) [inline .circleci/config.yml:22]: This should be checked first because it doesn't need any compilation

**Comment by iamareebjamal** [comment]
So now, there is no discrepancy between Android Studio formatting and spotless, right?

It shouldn't happen so that I import something in Android Studio and it autoformats imports wrongly and it fails the build

**Comment by adityastic** [comment]
@iamareebjamal yes, it matches android studios formatting completely now. Cloned and rechecked, works :+1:

File: .circleci/config.yml (hunk 2 above)
Comment (adityastic) [reply .circleci/config.yml:22]: On it

**Comment by iamareebjamal** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 15
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/fossasia/open-event-orga-app/pullRequest?prid=4677093&bid=15648597)

**Comment by iamareebjamal** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 15
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/fossasia/open-event-orga-app/pullRequest?prid=4677093&bid=15648597)

**Comment by iamareebjamal** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 15
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/fossasia/open-event-orga-app/pullRequest?prid=4677093&bid=15648597)

**Comment by iamareebjamal** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 15
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/fossasia/open-event-orga-app/pullRequest?prid=4677093&bid=15648597)

**Comment by iamareebjamal** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 15
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/fossasia/open-event-orga-app/pullRequest?prid=4677093&bid=15648597)

File: .circleci/config.yml [hunk 3]
```diff
@@ -5,7 +5,7 @@ jobs:
     docker:
       - image: circleci/android:api-28-alpha
     environment:
-      _JAVA_OPTIONS: -Xmx3600m
+      _JAVA_OPTIONS: -Xmx2048m -Dorg.gradle.daemon=false
```
Comment (iamareebjamal) [inline .circleci/config.yml:8]: Why decrease it?

Comment (adityastic) [reply .circleci/config.yml:8]: I tried dividing the workers into 4 with 3600 and it was still giving OOM error, so I tried decreasing the allowed memory and it worked

**Review by iamareebjamal — APPROVED** [review]

File: .circleci/config.yml (hunk 3 above)
Comment (iamareebjamal) [reply .circleci/config.yml:8]: Let's hope it continues to

**Pull request merged**